weblate/addons/git.py

```diff
@@ -60,6 +60,34 @@ def get_filenames(self, component):
                 languages[code].extend(translation.filenames)
         return languages
 
+    def get_squash_commit_message(self, repository, log_format, remote, filenames=None):
+        commit_message = self.instance.configuration.get("commit_message")
+
+        if not commit_message:
+            command = [
+                "log",
+                "--format={}".format(log_format),
+                "{}..HEAD".format(remote),
+            ]
+            if filenames:
+                command += ["--"] + filenames
+
+            commit_message = repository.execute(command)
+
+        if self.instance.configuration.get("append_trailers"):
+            command = [
+                "log",
+                "--format={}".format("%(trailers:separator)"),
```

Comment:
The formatting doesn't seem to be needed here.

This format was added in [Git 2.12.0](https://github.com/git/git/blob/master/Documentation/RelNotes/2.12.0.txt), so the git required version should be raised in `weblate/vcs/git.py`.